plugins/inputs/timex/timex.go

```diff
@@ -0,0 +1,94 @@
+//go:generate ../../../tools/readme_config_includer/generator
+//go:build linux
+
+package timex
+
+import (
+	_ "embed"
+	"fmt"
+	"time"
+
+	"golang.org/x/sys/unix"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Timex struct {
+	Log telegraf.Logger `toml:"-"`
+}
+
+func (*Timex) SampleConfig() string {
+	return sampleConfig
+}
+
+func (*Timex) Init() error {
+	return nil
+}
+
+func (*Timex) Gather(acc telegraf.Accumulator) error {
+	var timex = new(unix.Timex)
+
+	status, err := unix.Adjtimex(timex)
+	if err != nil {
+		return fmt.Errorf("failed to get time adjtimex stats: %w", err)
+	}
+
+	// Check the return status for clock state
+	// https://github.com/torvalds/linux/blob/master/include/uapi/linux/timex.h
+	syncStatus := 1
+	if status == unix.TIME_ERROR {
+		syncStatus = 0
+	}
```

Comment:
How about making this a boolean?

```suggestion
	synced := status != unix.TIME_ERROR
```